### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by Ladicek**
Motivation:

The context is https://github.com/vert-x3/issues/issues/633, and this commit is a first step towards that goal. It does:

- remove all the Hystrix stuff, because Hystrix is long dead
- improve documentation and implementation
- update test-scoped dependencies

I have also started working on a Resilience4j howto, which shall be linked from the documentation here. In fact, I already have Vert.x wrappers for all Resilience4j fault tolerance patterns (bulkhead, circuit breaker, rate limiter, retry, time limiter -- I didn't include cache for obvious reasons), so we could possibly make into a full-fledged extension (and deprecate the Vert.x circuit breaker maybe?). That's for later, for sure.

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -170,73 +178,10 @@ Each event contains a Json Object with:
...
-circuit breaker is allowed to execute the dangerous operation. Should the call succeed, the circuit breaker resets
+circuit breaker is allowed to execute the guarded operation. Should the call succeed, the circuit breaker resets
 and returns to the `closed` state, ready for more routine operation. If this trial call fails, however, the circuit
 breaker returns to the `open` state until another timeout elapses.
-
-== Reported exceptions
-
-The fallback receives a:
-
-* {@link io.vertx.circuitbreaker.OpenCircuitException} when the circuit breaker is opened
```
Comment (vietj) [inline src/main/asciidoc/index.adoc:185]: Why is this section about reported exception deleted ?

**Review by vietj — COMMENTED** [review]
minor concern about the exception part deleted

File: src/main/asciidoc/index.adoc (hunk 1 above)
Comment (Ladicek) [reply src/main/asciidoc/index.adoc:185]: It isn't deleted, it's moved up a little: https://github.com/vert-x3/vertx-circuit-breaker/pull/71/commits/41fd1cfaa6fcf964305be1e4dcd1bc000df487a9#diff-97965d9bb2e3f738e48945f14914e26a4f3c5ae77cfeb5f339009e191124f2beR88

**Comment by Ladicek** [comment]
Added one more commit that mentions Resilience4j at the end of the documentation (where Hystrix usage used to be documented). The documentation is very brief, it mostly just points to the recently published how-to: https://how-to.vertx.io/resilience4j-howto/

**Comment by tsegismont** [comment]
> I have also started working on a Resilience4j howto, which shall be linked from the documentation here. In fact, I already have Vert.x wrappers for all Resilience4j fault tolerance patterns (bulkhead, circuit breaker, rate limiter, retry, time limiter -- I didn't include cache for obvious reasons), so we could possibly make into a full-fledged extension (and deprecate the Vert.x circuit breaker maybe?). That's for later, for sure.

Another option would be a Reactiverse project for Resilience4j integration

**Comment by vietj** [comment]
can you rebase this PR please @Ladicek ?

**Comment by Ladicek** [comment]
Rebased.

**Comment by vietj** [comment]
Thanks @Ladicek

**Pull request merged**